network/p2p/pubsub.go

```diff
@@ -98,11 +99,34 @@ func makePubSub(ctx context.Context, cfg config.Local, host host.Host) (*pubsub.
 		pubsub.WithMessageSignaturePolicy(pubsub.StrictNoSign),
 		// pubsub.WithValidateThrottle(cfg.TxBacklogSize),
 		pubsub.WithValidateWorkers(incomingThreads),
+		pubsub.WithEventTracer(&pubsubEventTracer{log: log}),
 	}
 
 	return pubsub.NewGossipSub(ctx, host, options...)
 }
 
+type pubsubEventTracer struct {
```

Comment:
Assume we do not intend to merge this right?

Reply:
yes, plan to run it few times to obtain traces on failure